editor/scene_tree_dock.cpp

```diff
@@ -1774,14 +1776,7 @@ void SceneTreeDock::perform_node_renames(Node *p_base, HashMap<Node *, NodePath>
 							if (found_path->value.is_empty()) {
 								//will be erased
 
-								int idx = 0;
-								HashSet<int>::Iterator EI = ran.begin();
-								ERR_FAIL_COND(!EI); //bug
-								while (*EI != i) {
-									idx++;
-									++EI;
-									ERR_FAIL_COND(!EI); //another bug
-								}
+								const int idx = i - tracks_removed++;
```

Comment:
Just as style suggestion: this doesn't need to be `const`, the block is pretty short, and also we prefer to make things explicit, so the increment should be moved to a separate line.